### PR TITLE
feat: Added validate and validateSync function

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: src/validate_async.cc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    uv_work_t *request = ::new uv_work_t();
+    request->data = baton;
+
+    // Schedule the work request
+    int status = uv_queue_work(uv_default_loop(),
+                                request,
+                                AsyncValidate,
+                                (uv_after_work_cb)AsyncValidateAfter);
+
+    assert(status == 0);
```
Comment (w-vi) [inline src/validate_async.cc]: are you aware that the assert is removed from the release build and so there is effectively no check for status in release version?

**Comment by w-vi** [comment]
Apart from that comment on `assert` it LGTM.

File: src/validate_async.cc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    return;
+}
+
+void AsyncValidate(uv_work_t* request) {
+    Baton* baton = static_cast<Baton*>(request->data);
+
+    // Validate the source data
+    drafter_result* parseResult = drafter_check_blueprint(baton->sourceData.c_str());
```
Comment (w-vi) [inline src/validate_async.cc]: What happens when it returns NULL as is the case when there are no warnings and errors?

File: src/validate_async.cc (hunk 1 above)
Comment (pksunkara) [inline src/validate_async.cc]: Non debugging build of node.js? or protagonist?

Comment (w-vi) [inline src/validate_async.cc]: protagonist

Comment (w-vi) [inline src/validate_async.cc]: return an error in the callback?

Comment (pksunkara) [inline src/validate_async.cc]: I think this should be a different PR since parseSync and parse need to changed too.

File: src/validate_sync.cc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        free(optionsResult);
+    }
+
+    drafter_parse_options drafterOptions;
+
+    drafterOptions.requireBlueprintName = options & snowcrash::RequireBlueprintNameOption;
+
+    // Validate the source data
+    drafter_result* parseResult = drafter_check_blueprint_with_options(*sourceData, drafterOptions);
+
```
Comment (w-vi) [inline src/validate_sync.cc]: I don't understand you can easily have
```cpp
if (parseResult != NULL) {
    drafter::WrapperOptions wrapperOptions(drafter::RefractASTType);
    drafter::ConversionContext context(wrapperOptions);
 
    sos::Object sosResult = SerializeRefract(parseResult, context);
    info.GetReturnValue().Set(v8_wrap(sosResult)->ToObject());
} else {
    info.GetReturnValue().Set(Nan::Null());
}
